tensorflow_datasets/core/custom_dataset/image_folder.py

```
@@ -0,0 +1,168 @@
+
+  ```
+  path/to/manual_dir/<dataset_name>/
+    split_name/  # Ex: 'train'
+      label1/  # Ex: 'airplane' or '0015'
+        xxx.png
+        xxy.png
+        xxz.png
+      label2/
+        xxx.png
+        xxy.png
+        xxz.png
+    split_name/  # Ex: 'test'
+      ...
+  ```
+
+  To use it:
+
+  ```
+  builder = tfds.ImageLabelFolder(root_dir='path/to/manual_dir/')
+  print(builder.info)  # Splits, num examples,... automatically calculated
+  ds = builder.as_dataset(split='split_name', shuffle_files=True)
+  ```
+
+  """
+
+  VERSION = core.Version('2.0.0')
+
+  def __init__(self, root_dir):
+    self._version = str(self.VERSION)
+    self._root_dir = os.path.expanduser(root_dir)
+
+    # dict[split_name][label_name] = list(img_paths)
+    self._split_label_images, labels = _get_split_label_images(self._root_dir)
+
+    self._ds_info = core.DatasetInfo(
+        builder=self,
+        description='Generic image classification dataset.',
+        features=core.features.FeaturesDict({
+            'image': core.features.Image(),
+            'label': core.features.ClassLabel(names=sorted(list(labels))),
+        }),
+        supervised_keys=('image', 'label'),
+    )
+
+    super(ImageLabelFolder, self).__init__(
+        data_dir=self._root_dir, version=self._version, config=None)
+
+    # Calculate splits for DatasetInfo
+    splits = {
+        split_name: sum(len(l)
+                        for l in self._split_label_images[split_name].values())
+        for split_name in self._split_label_images
+    }
+
+    split_dict = core.SplitDict(self.name)
+    for split_name, size in splits.items():
+      split_dict.add(
+          core.SplitInfo(name=split_name, shard_lengths=[size]))
+
+    self._ds_info.update_splits_if_different(split_dict)
+
+    # Reset `_data_dir` as it should not change to DATA_DIR/Version
+    self._data_dir = self._root_dir
+
+  def _info(self):
+    return self._ds_info
+
+  def _download_and_prepare(self, **kwargs):
+    pass
```

We should raise an error here.
Also could you add unittest to make sure the error is correctly raised ?

If this was changed for the test, then the test should be modified to make this a no-op